runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -221,55 +223,30 @@ private void drawTooltipIfMouseover(Graphics2D graphics, XpGlobe mouseOverSkill,
 		int y = (int) drawnGlobe.getY() + DEFAULT_CIRCLE_HEIGHT + 10;
-		int padding = (TOOLTIP_RECT_SIZE_X - TOOLTIP_TEXT_RECT_SIZE_X) / 2;
-		int stringX = x + padding;
 
+		String skillName = mouseOverSkill.getSkillName();
 		String skillLevel = Integer.toString(mouseOverSkill.getCurrentLevel());
-		String skillCurrentXp = Integer.toString(mouseOverSkill.getCurrentXp());
-		String skillXpToLvl = Integer.toString((mouseOverSkill.getGoalXp() - mouseOverSkill.getCurrentXp()));
-
-		FontMetrics fm = graphics.getFontMetrics();
-		int skillLevelX = x + padding + (TOOLTIP_TEXT_RECT_SIZE_X - fm.stringWidth(skillLevel));
-		int skillCurrentXpX = x + padding + (TOOLTIP_TEXT_RECT_SIZE_X - fm.stringWidth(skillCurrentXp));
-		int skillXpToLvlX = x + padding + (TOOLTIP_TEXT_RECT_SIZE_X - fm.stringWidth(skillXpToLvl));
-		int stringHeight = fm.getHeight();
-
-		//draw tooltip container
-		graphics.setPaint(DEFAULT_XPGLOBE_BACKGROUND_COLOR);
-		graphics.fillRect(x, y, TOOLTIP_RECT_SIZE_X, TOOLTIP_RECT_SIZE_Y);
-		graphics.setPaint(DEFAULT_PROGRESS_REMAINDER_ARC_COLOR);
-		graphics.setStroke(new BasicStroke(2));
-		graphics.drawRect(x, y, TOOLTIP_RECT_SIZE_X, TOOLTIP_RECT_SIZE_Y);
-
-		//draw the text
-		graphics.setPaint(Color.WHITE);
-		graphics.drawString(mouseOverSkill.getSkillName(), stringX, y + stringHeight);
-		graphics.drawString(skillLevel, skillLevelX, y + stringHeight);
-		graphics.drawString("Current exp:", stringX, y + (stringHeight * 2));
-		graphics.drawString(skillCurrentXp, skillCurrentXpX, y + (stringHeight * 2));
-		graphics.drawString("Exp to level:", stringX, y + (stringHeight * 3));
-		graphics.drawString(skillXpToLvl, skillXpToLvlX, y + (stringHeight * 3));
-
-		//draw the progress bar
-		double progress = mouseOverSkill.getSkillProgress(Experience.getXpForLevel(mouseOverSkill.getCurrentLevel()), mouseOverSkill.getCurrentXp(), mouseOverSkill.getGoalXp());
-		int barWidth = TOOLTIP_TEXT_RECT_SIZE_X;
-		int barHeight = 16;
-		int barX = x + padding;
-		int barY = y + (stringHeight * 3) + 10;
-
-		DecimalFormat df = new DecimalFormat("#.00");
-		String progressText = df.format(progress) + "%";
-		int progressTextLength = fm.stringWidth(progressText);
-		int progressTextX = barX + (barWidth / 2) - (progressTextLength / 2);
-		int progressTextY = barY + 12;
-
-		int progressFill = (int) ((barWidth / 100F) * progress);
-
-		graphics.setColor(Color.WHITE);
-		graphics.fillRect(barX, barY, barWidth, barHeight);
-		graphics.setColor(Color.GREEN);
-		graphics.fillRect(barX, barY, progressFill, barHeight);
-		graphics.setPaint(Color.BLACK);
-		graphics.drawString(progressText, progressTextX, progressTextY);
+
+		DecimalFormat decimalFormat = new DecimalFormat("###,###,###");
+		String skillCurrentXp = decimalFormat.format(mouseOverSkill.getCurrentXp());
+		String skillXpToLvl = decimalFormat.format(mouseOverSkill.getGoalXp() - mouseOverSkill.getCurrentXp());
+
+		PanelComponent xpTooltip = new PanelComponent();
+		xpTooltip.getLines().clear();
```

Comment:
No need to clear the lines here, as it is fresh object the lines are already empty.